### PR TITLE
Pyscript blog

### PR DESCRIPTION
**Pull request by alan-cugler**
Davids blog post on pyscript still being processed.

### CODE REVIEW

**Comment by alan-cugler** [comment]
@toonarmycaptain I think the ending paragraphs need to be broken up. you have a couple really sick lines I love and they are lsot in the wall of text. so we need to break those up and emphasize them.

**Comment by alan-cugler** [comment]
maybe we can add these page breaks for the separate sections of the blog: https://docs.google.com/document/d/1qHrVi9BxpLsY72BCLWqAA0YuXo93-RoQkFh3U5UCuZY/edit?usp=sharing

**Comment by toonarmycaptain** [comment]
> probably going to ask Connor to give the live pyscript bit a new CSS

Inside the iFrame, or within in the page itself? I think the latter is cooler, but not sure of the lift required.

**Comment by toonarmycaptain** [comment]
> Mike would like you to attempt to make that actual visual for pyscript graphing doesnt need to be the literal graphic you have there, but try to generate one from actual pyscript stuff.

Got a slick TL logo or something for the backing?

**Comment by alan-cugler** [comment]
https://docs.google.com/document/d/1qHrVi9BxpLsY72BCLWqAA0YuXo93-RoQkFh3U5UCuZY/edit?usp=sharing
These are the page breaks I was considering for separating chunks of your blog

**Comment by alan-cugler** [comment]
> > probably going to ask Connor to give the live pyscript bit a new CSS
> 
> Inside the iFrame, or within in the page itself? I think the latter is cooler, but not sure of the lift required.

We are going to make copies of the pyscript.js and pyscript.css file and resolve CSS class conflicts and then self host them on this repo.

**Comment by toonarmycaptain** [comment]
@alan-cugler I've broken up that text. If you can get those TL graphics to punctuate breaks, we can insert them. 
I'm thinking some bolded text for emphasis might work in spots too.

File: content/blog/pyscript-intro/contents.lr [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+<script defer src="https://pyscript.net/alpha/pyscript.js"></script>
+
+When working with the frontend, logic is written in Javascript. However, most kids, and a lot of backend engineers learn and use python as their primary language. This, coupled with machine learning and datascience being dominated by python, now we can bring a frontend logic alternative using PyScript. 
+
+Fear js no more - PyScript is here. 
+
+
+PyScript is a framework that allows python code to run in the browser with near-native speed.
+
+This is accomplished with a toolchain of Pyodide-Emscripten-WASM. Pyodide takes CPython, adds an interface between python and Typescript/JavaScript, and using Emscripten compiles it down to WASM (WebAssembly). WASM binaries executes at near-native speed and is supported by all major browsers.
```
Comment (toonarmycaptain) [inline content/blog/pyscript-intro/contents.lr:40]: Do we want links for any of these tools?

**Comment by toonarmycaptain** [comment]
Should we pin and bundle the pyscript binary as well? That's probably good practice, aside from guarding against any changes that break our current demos?

**Comment by toonarmycaptain** [comment]
Were we interested in moving that greyed out loading screen to something less intrusive, maybe taking the grey out and moving the loading icon to a bottom corner of the screen?

**Comment by nixjdm** [comment]
@toonarmycaptain Yes, we should vendor every artifact from pyscript that we're using to ensure this post works no matter what links fail in the future, or changes being made to artifacts at the same links. Also, the disclaimer at the bottom should have an expandable code block to show the diff between our css and the original.

If you can get rid of the whole-page loading screen, or effectively hide it, that would be awesome. I was actually thinking we could make another blog post specifically about how to do that, but if you have a solution now, we can use it here on this one.

File: content/blog/pyscript-intro/contents.lr [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+
+print('Here is your plot:')
+fig # Like print() this will display the plot.
+</py-script>
+
+<div id="plot_display" align="center"></div>
+</div>
+
+
+Yep, you can pull your own code along for the ride and import it inside your `<py-script>` tags. You can actually bundle files of any type and use them in your code.
```
Comment (nixjdm) [inline content/blog/pyscript-intro/contents.lr:218]: Right above this line, I get the error:

> JsException(PythonError: Traceback (most recent call last): File "/lib/python3.10/site-packages/_pyodide/_base.py", line 421, in eval_code CodeRunner( File "/lib/python3.10/site-packages/_pyodide/_base.py", line 237, in __init__ self.ast = next(self._gen) File "/lib/python3.10/site-packages/_pyodide/_base.py", line 141, in _parse_and_compile_gen mod = compile(source, filename, mode, flags | ast.PyCF_ONLY_AST) File "", line 4 Get the data ^^^ SyntaxError: invalid syntax )


Screenshot from 2022-08-29 15-06-02

Comment (toonarmycaptain) [reply content/blog/pyscript-intro/contents.lr:218]: I'm not sure what this is. I tried a fresh clone, fresh env, different browser.

Comment (nixjdm) [reply content/blog/pyscript-intro/contents.lr:218]: Try clearing your browser cache? I haven't got this section to work yet, on firefox or chrome.

Comment (toonarmycaptain) [reply content/blog/pyscript-intro/contents.lr:218]: Oddness. I tried clearing caches as well. I installed Opera and tried that, it worked. 
Did the # in front of `Get the data` somehow get removed? Or some hidden indentation added?

Comment (BenjaminEchelmeier) [reply content/blog/pyscript-intro/contents.lr:218]: I managed to replicate the error--by deleting the # and return before "Get the data," and I get the exact error code.

**Comment by toonarmycaptain** [comment]
> If you can get rid of the whole-page loading screen, or effectively hide it, that would be awesome. I was actually thinking we could make another blog post specifically about how to do that, but if you have a solution now, we can use it here on this one.

I don't know how to do it, but I believe @BenjaminEchelmeier saw it in the CSS or something.

File: content/blog/pyscript-intro/contents.lr (hunk 2 above)
Comment (toonarmycaptain) [reply content/blog/pyscript-intro/contents.lr:218]: Would we be fine moving forward with the comments only being present in the displayed code (if we need to)? That should obliviate the issue, and the code with comments should work in most other contexts, it not being an issue with pyscript.

**Comment by nixjdm** [comment]
@toonarmycaptain I still would like you to vendor all external pyscript deps before pulling, like we are for the css.

With the note about the custom css, I'd also like you to please include the actual diff between the files, and add a brief sentence saying explicitly which version of pyscript we're using so other's can reproduce on their own, even if that's just "latest as of <date>".

Finally, this branch also has conflicts that need resolved before merging is possible.

**Comment by nixjdm** [comment]
Note to merger - squash merge this PR. The git history's too long to drag it along.

**Comment by BenjaminEchelmeier** [comment]
I vendored the pyscript from /latest to freeze it, but I'm having trouble verifying what version it is. They do have a download button on the website with instructions to then point to the pyscript.js, but it doesn't look like there is a pyscript.js file in to download.
Also, we were using the same dir name as the other pyscript blog, so I changed it to avoid conflicts.

**Comment by BenjaminEchelmeier** [comment]
Okay, found the version and dropped some notes on our changes. I think we're pretty well good-to-go.

**Comment by toonarmycaptain** [comment]
> I vendored the pyscript from /latest to freeze it, but I'm having trouble verifying what version it is. 
They've got a [releases feed in the pyscript repo](https://github.com/pyscript/pyscript/releases) if you want a link to reference that one might assume won't be as vulnerable to going stale as the website. 

One might argue that taking that asset (which has a pyscript.ts, but I don't think that's the bundle that you're getting when you use the `latest` link in a page) and building it yourself might be better, but that's really a whole other blogpost and whole matrix of dependencies to pin down.

**Comment by alan-cugler** [comment]
image

**Comment by alan-cugler** [comment]
I got one error and other than that the blog was in good shape

**Comment by toonarmycaptain** [comment]
> image

Looks like the comment after the print fig `fig # Like print() this will display the plot.` is the issue. 
It's probably an issue with [mistune](https://github.com/lepture/mistune/issues/315).

**Comment by toonarmycaptain** [comment]
Nevermind...for some reason the link to the data source in the code (ie the functioning py-script tag) has angle brackets around it. This *actually* fixed it.

**Pull request merged**